### PR TITLE
fix(brain): harden lesson consolidation follow-up

### PR DESCRIPTION
**Pull request by djm204**
## Summary
- harden repeated-failure lesson consolidation identified by post-merge Codex review on #3760
- preserve restart-safe lineage and never-store behavior across encrypted/plaintext brains
- bound relevance pagination and validate consolidation/storage edge cases

## Verification
- `npm test --workspace @franken/brain` (425 tests passed)
- `npm run lint --workspace @franken/brain`
- `npm run typecheck --workspace @franken/brain`
- `npm run build --workspace @franken/brain`
- `npm test --workspace @franken/types`
- `npm run lint --workspace @franken/types`
- `npm run typecheck --workspace @franken/types`
- `npm run build --workspace @franken/types`
- `git diff --check`

Follow-up to #3760.

### CODE REVIEW

**Comment by djm204** [comment]
@codex review

File: packages/franken-brain/src/sqlite-brain.ts [hunk 1]
```diff
@@ -6712,26 +6798,43 @@ export class SqliteBrain implements IBrain {
...
+            key: candidate.key,
+            status,
+            relevance: lessonRelevance(normalizedQuery, value),
+          } satisfies RelevantLesson;
+          if (lesson.relevance <= 0 || lesson.confidence < minConfidence) continue;
+          lessons.push(lesson);
+          matchedKeys.add(candidate.key);
+          if (lessons.length >= limit) return lessons;
+        }
+        if (page.length < pageSize) break;
```
[the remark this answers is not in the document]
Comment (djm204) [reply packages/franken-brain/src/sqlite-brain.ts:6871]: Addressed in the latest pushed fixes; resolving this Codex thread and requesting a fresh review.

File: packages/franken-brain/src/sqlite-brain.ts [hunk 2]
```diff
@@ -5767,31 +5785,52 @@ export class SqliteMemoryReviewQueue {
...
-      `SELECT id, value FROM memory_review_candidates
+      `SELECT id, value, candidate_kind, lesson_evidence_ids, lesson_suppression_tokens
+         FROM memory_review_candidates
        WHERE candidate_kind IS NULL
           OR (candidate_kind = 'consolidated-lesson'
-              AND (lesson_evidence_ids IS NULL OR lesson_suppression_tokens IS NULL))`,
-    ).all() as Array<{ id: string; value: string }>;
+              AND (${this.encryption
+                ? '1 = 1'
+                : 'lesson_evidence_ids IS NULL OR lesson_suppression_tokens IS NULL'}))`,
```
[the remark this answers is not in the document]
Comment (djm204) [reply packages/franken-brain/src/sqlite-brain.ts:5805]: Addressed in the latest pushed fixes; resolving this Codex thread and requesting a fresh review.

File: packages/franken-brain/src/sqlite-brain.ts [hunk 3]
```diff
@@ -6131,20 +6182,41 @@ function parseLessonSuppressionTokens(value: string | null | undefined): string[
...
     ...(lesson.searchTerms ?? []),
   ])]
-    .map((token) => createHash('sha256').update(token).digest('hex'))
+    .flatMap((token) => {
+      const dictionaryHash = createHash('sha256').update(token).digest('hex');
+      if (!encryption) return [dictionaryHash];
+      return [
+        `h1:${encryption.fingerprint('lesson-suppression-token:v1', token)}`,
+        `h0:${encryption.fingerprint('lesson-suppression-token-sha256:v1', dictionaryHash)}`,
+      ];
```
[the remark this answers is not in the document]
Comment (djm204) [reply packages/franken-brain/src/sqlite-brain.ts:6218]: Addressed in the latest pushed fixes; resolving this Codex thread and requesting a fresh review.

File: packages/franken-brain/src/sqlite-brain.ts [hunk 4]
```diff
@@ -6648,18 +6726,26 @@ export class SqliteBrain implements IBrain {
...
         }
         if (candidate.status === 'pending' && isConsolidatedLesson(candidate.value)) {
           pendingLessons.push(candidate);
-          created.push({ id: candidate.id, key: candidate.key, status: 'pending', value: candidate.value });
+          created.push({
+            id: candidate.id,
+            key: candidate.key,
+            status: 'pending',
+            value: candidate.value,
+            ...(candidate.replaces ? { replaces: candidate.replaces } : {}),
```
[the remark this answers is not in the document]
Comment (djm204) [reply packages/franken-brain/src/sqlite-brain.ts:6785]: Addressed in the latest pushed fixes; resolving this Codex thread and requesting a fresh review.

**Comment by djm204** [comment]
@codex review

File: packages/franken-brain/src/sqlite-brain.ts [hunk 5]
```diff
@@ -6712,26 +6836,44 @@ export class SqliteBrain implements IBrain {
...
+            ...value,
+            confidence: candidate.confidence,
+            key: candidate.key,
+            status,
+            relevance: lessonRelevance(normalizedQuery, value),
+          } satisfies RelevantLesson;
+          if (lesson.relevance <= 0 || lesson.confidence < minConfidence) continue;
+          lessons.push(lesson);
+          matchedKeys.add(candidate.key);
+          if (lessons.length >= limit) return lessons;
```
[the remark this answers is not in the document]
Comment (djm204) [reply packages/franken-brain/src/sqlite-brain.ts:6869]: Deferred under the two-Codex-round cap: these independent P2 findings do not block the epic closeout. Filed granular follow-ups: ranking-before-limit #3768, conflict-resolution retirement #3769, and atomic durable deletion #3770. No further @codex review trigger will be issued for this PR.

File: packages/franken-brain/src/sqlite-brain.ts [hunk 6]
```diff
@@ -6648,18 +6764,26 @@ export class SqliteBrain implements IBrain {
           reason: `${mergedOccurrenceCount} similar failure episodes formed an expanded lesson revision for review.`,
         });
         if (candidate.status === 'pending') {
+          const replacements = matchingApproved.slice(1).map(({ candidate: matched }) => ({
+            key: matched.key,
+            candidateId: matched.id,
+          }));
           this.memoryReview.setLessonReplacementKeys(
```
[the remark this answers is not in the document]
Comment (djm204) [reply packages/franken-brain/src/sqlite-brain.ts:6771]: Deferred under the two-Codex-round cap: these independent P2 findings do not block the epic closeout. Filed granular follow-ups: ranking-before-limit #3768, conflict-resolution retirement #3769, and atomic durable deletion #3770. No further @codex review trigger will be issued for this PR.

File: packages/franken-brain/src/sqlite-brain.ts [hunk 7]
```diff
@@ -4214,6 +4228,7 @@ export class SqliteMemoryReviewQueue {
     if (retiredLessonKeys.length > 0) {
       for (const key of retiredLessonKeys) this.working.delete(key);
       this.working.flushToDb();
+      this.expireWorkingKeys?.(retiredLessonKeys);
```
[the remark this answers is not in the document]
Comment (djm204) [reply packages/franken-brain/src/sqlite-brain.ts:4231]: Deferred under the two-Codex-round cap: these independent P2 findings do not block the epic closeout. Filed granular follow-ups: ranking-before-limit #3768, conflict-resolution retirement #3769, and atomic durable deletion #3770. No further @codex review trigger will be issued for this PR.
